src/components/Chart/data/dataSelector.ts

```diff
@@ -8,17 +8,20 @@
 
 import { options, timestampToIndex } from '../../../globals';
 import bitDataSelector from './bitDataSelector';
+import { createEmptyArrayWithAmpereState } from './commonBitDataFunctions';
+import { DigitalChannelsType } from './dataTypes';
 import noOpBitDataProcessor from './noOpBitDataProcessor';
 
-const emptyArray = () =>
-    [...Array(4000)].map(() => ({ x: undefined, y: undefined }));
-
 export default () => ({
-    ampereLineData: emptyArray(),
+    ampereLineData: createEmptyArrayWithAmpereState(),
     bitDataSelector: bitDataSelector(),
     noOpBitDataProcessor: noOpBitDataProcessor(),
 
-    process(begin, end, digitalChannelsToCompute) {
+    process(
+        begin: number,
+        end: number,
```

Comment:
@datenreisender , I think you mentioned that it would be nice if this were either typed with index or timestamp according to if they are index or type, since this isn't always obvious. Therefore, I wonder if we should have created specific types for e.g. `sampleIndex` and `sampleTimestamp`, which are basically `number` type?

Reply:
Yes. Maybe also make types for things like a rate and a duration in microseconds, so it is something like this:

```ts
type rate = number;
type microseconds = number;
type sampleIndex = number;
type timestamp = number;

interface GlobalOptions {
    samplingTime: microseconds;
    samplesPerSecond: rate;
    currentIndex: sampleIndex;
    currentTimestamp: timestamp | undefined | null;
}
```

These types sometimes get lost and then TypeScript again tells you that it is just a `number` and it doesn't catch any type errors (you can still say e.g. `options.samplingTime = options.samplesPerSecond`) but it still gives you some hints about the intention.